processing/src/main/java/org/apache/druid/timeline/CompactionState.java

```diff
@@ -50,6 +59,26 @@
  */
 public class CompactionState
 {
+
+  /**
+   * Lazy initialization holder for deterministic ObjectMapper.
```

Comment:
I wonder if we shouldn't just inject this mapper annotated with `@Sorted` or `@Deterministic` as a lazy singleton. It may be injected into `CompactionStateManager` and fingerprints will always be created by that class rather than using a static utility method.